Source/Core/Core/IOS/Network/KD/NetKDRequest.cpp

```diff
@@ -183,6 +201,78 @@ void NetKDRequestDevice::Update()
   }
 }
 
+void NetKDRequestDevice::SchedulerTimer()
+{
+  u32 mail_time_state = 0;
+  u32 download_time_state = 0;
+  Common::SetCurrentThreadName("KD Scheduler Timer");
+  while (true)
+  {
+    {
+      std::lock_guard lg(m_scheduler_lock);
+      if (m_mail_span <= mail_time_state)
+      {
+        m_scheduler_work_queue.EmplaceItem([this] { SchedulerWorker(SchedulerEvent::Mail); });
+        INFO_LOG_FMT(IOS_WC24, "NET_KD_REQ: Dispatching Mail Task from Scheduler");
+        mail_time_state = 0;
+      }
+
+      if (m_download_span <= download_time_state)
+      {
+        INFO_LOG_FMT(IOS_WC24, "NET_KD_REQ: Dispatching Download Task from Scheduler");
+        m_scheduler_work_queue.EmplaceItem([this] { SchedulerWorker(SchedulerEvent::Download); });
+        download_time_state = 0;
+      }
+    }
+
+    if (m_shutdown_event.WaitFor(std::chrono::minutes{1}))
+      return;
+
+    mail_time_state++;
+    download_time_state++;
+  }
+}
+
+void NetKDRequestDevice::SchedulerWorker(const SchedulerEvent event)
+{
+  if (event == SchedulerEvent::Download)
+  {
+    // TODO: Implement downloader part of scheduler
+    return;
+  }
+  else
+  {
+    if (!m_config.IsRegistered())
+      return;
+
+    u32 mail_flag{};
+    u32 interval{};
+
+    NWC24::ErrorCode code = KDCheckMail(&mail_flag, &interval);
+    if (code != NWC24::WC24_OK)
+    {
+      LogError(ErrorType::CheckMail, code);
+    }
+  }
+}
+
+std::string_view NetKDRequestDevice::GetValueFromCGIResponse(const std::string& response,
+                                                             const std::string& key)
+{
+  const std::vector<std::string> raw_fields = SplitString(response, '\n');
+  for (const std::string& field : raw_fields)
+  {
+    const std::vector<std::string> key_value = SplitString(field, '=');
+    if (key_value.size() != 2)
+      continue;
+
+    if (key_value[0] == key)
+      return StripWhitespace(key_value[1]);
```

Comment:
Here you are returning a string_view from a temporary, I think that's going to be a problem depending on how the memory is managed.

Reply:
Tested and it was a problem, switched back to `string`